### PR TITLE
web: add unit test to cover `Panel` location change

### PR DESCRIPTION
**Pull request by valerybugakov**
## Changes

- Added unit test to cover location mutation on tab change in the `Panel` component.

Follow up for https://github.com/sourcegraph/sourcegraph/pull/22600.

### CODE REVIEW

**Review by umpox — APPROVED** [review]
Nice!

File: client/shared/src/testing/render-wth-router.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { RenderResult, render } from '@testing-library/react'
```
Comment (pdubroy) [inline client/shared/src/testing/render-wth-router.tsx:1]: Nit: typo in the filename (`wth`)?

File: client/shared/src/testing/render-wth-router.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { MemoryHistory, createMemoryHistory } from 'history'
+import React, { ReactNode } from 'react'
+import { Router } from 'react-router-dom'
+
+export interface RenderWithRouterReturn extends RenderResult {
```
Comment (pdubroy) [inline client/shared/src/testing/render-wth-router.tsx:6]: Nit: maybe `RenderWithRouterResult` is a more appropriate name, given that it extends `RenderResult`?

**Review by pdubroy — APPROVED** [review]
lgtm, thanks!

File: client/shared/src/testing/render-wth-router.tsx (hunk 1 above)
Comment (valerybugakov) [reply client/shared/src/testing/render-wth-router.tsx]: Oh, thanks!

**Pull request merged**